onnxruntime/contrib_ops/cpu/gather_nd.cc

```diff
@@ -45,12 +46,16 @@ Status GatherNDBase::PrepareForCompute(OpKernelContext* context, Prepare& p) con
   auto output_tensor = context->Output(0,TensorShape(shape));
   std::vector<int64_t> element_counts(last_indice_dimension, 0LL); // Number of elements for each input dimension
 
-#ifdef USE_OPENMP
-#pragma omp parallel for
-#endif
-  for (int64_t i = 0; i < last_indice_dimension; ++i) {
-    element_counts[i] = input_shape.SizeFromDimension(i + 1);
-}
+  auto ctx_internal = static_cast<OpKernelContextInternal*>(context);
+  auto thread_pool = const_cast<concurrency::ThreadPool*>(ctx_internal->GetOperatorThreadPool());
+
```

Comment:
If the ThreadPool instance needs to be non-const to be used, should GetOperatorThreadPool return a non-const* instead of having const_cast everywhere?

Reply:
Several people requested that the GetOperatorThreadPool be lifted to the main context or session.  I left that change to be done when the decision had been made concerning where to move the API.